partner_code_program_purchase/tests/test_partner_code_program_purchase.py

```diff
@@ -0,0 +1,35 @@
+from odoo.tests.common import TransactionCase
+
+
+class TestPartnerCodeProgramPurchase(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        cls.partner_code_program = cls.env["res.partner.code.program"].create(
+            {
+                "name": "Program test",
+                "code": "CP-001",
+            }
+        )
+
+        cls.partner = cls.env["res.partner"].create(
+            {
+                "name": "Partner test",
+                "code_program_id": cls.partner_code_program.id,
+            }
+        )
+
+        cls.purchase_report = cls.env["purchase.report"]
+
+    def test_code_program_id(self):
+        self.assertIn("code_program_id", self.purchase_report._fields)
+
+    def test_select(self):
+        self.assertIn(
+            "partner.code_program_id as code_program_id",
+            str(self.purchase_report._select()),
+        )
+
+    def test_group_by(self):
+        self.assertIn("partner.code_program_id", str(self.purchase_report._group_by()))
```

Comment:
I think this tests are too simple, and apart from increasing the codecov they do not apport too much. But keep them